docker/README.md

```diff
@@ -0,0 +1,136 @@
+
+- `COW`
+
+  *NOTE* `MOCK` and `COW` have some weird requirements tied to assumptions in the [packaging repo](https://github.com/puppetlabs/packaging). There's a plan to make this better at some point.
+
+  The deb platforms you want to build. Individual entries must match the pattern
+  `base-<codename>-<arch>.cow`. For multiple OSes, use a space separated list. Defaults to 
+  `base-jessie-i386.cow base-xenial-i386.cow base-stretch-i386.cow base-bionic-i386.cow`
+
+- `UPDATE_EZBAKE_VERSION`
+
+  Whether or not to update the ezbake version in the project you're building. This is
+  mostly intended to let you test ezbake changes without having to commit code with a `-SNAPSHOT`
+  version in project.clj. Defaults to false
+
+- `EZBAKE_VERSION`
+
+  The version of ezbake to use for your build. Only does anything if `UPDATE_EZBAKE_VERSION`
+  is set to true. Defaults to unset
+
+
+## Running
+
+There are a few different ways I've found for running builds on this container. I'll document
+the pros and cons here.
+
+This container assumes that the code you're building will be in `/workspace`. The packages
+will be copied into `/output` before the container terminates, and if you're building a
+custom ezbake that will be cloned into `/ezbake`.
+
+1. Run from a remotely accessible repo (github, gitlab, etc)
+
+This is by far the fastest build option, but does require the extra step of committing and pushing your code
+rather than using the code in your working dir.
+
+```
+docker run --rm --volume $(PWD)/output:/output --env PROJECT_REPO=https://github.com/underscorgan/puppetdb --env PROJECT_BRANCH=my_feature_branch puppet/ezbake
+```
+
+2. Mount in your current working directory
+
+    * Please note: due to the amount of I/O used in compiling the artifacts it will be _incredibly_ slow if you
+mount in your entire current working directory. The options below will speed it up significantly (I haven't
+actually managed to get a build to finish and have killed it after ~30 minutes when I've mounted in my whole
+working directory), but it still takes significantly more than the builds from remote repo. The options are:
+
+    a. Mount in full working directory and use anonymous volumes for the I/O heavy directories
+
+    This is the slower of the two options, but also the simplest. The volumes in the example should be the
+    same for all ezbake projects.
+
+    ```
+    docker run --rm --volume $(PWD)/output:/output --volume /workspace/test --volume /workspace/test-resources --volume /workspace/target --volume /workspace/tmp puppet/ezbake
+    ```
+    
+    b. Only mount in needed files/directories
+
+    This is faster than mounting in everything and using anonymous volumes, but it'll take more effort
+    to get the correct things mounted in. This will vary some between projects, but you can get a pretty
+    good idea of what is needed by looking at the `.gitignore` or `.dockerignore` files.
```

Comment:
I assume the things in those files are things you DON'T need? That could be made clearer..

Reply:
oh wait, I see - the .dockerignore says to ignore everything _except_ those things.. Maybe only reference that, instead of the .gitignore? since they're following different patterns here.

Reply:
I included both of them in case the `.dockerignore` file doesn't exist, since you may be building a project (or a branch) that doesn't have a corresponding container.

Reply:
ookay!